packages/twenty-server/src/modules/messaging/types/gmail-message-parsed-response.ts

```diff
@@ -1,12 +1,6 @@
-export type GmailMessageParsedResponse = {
-  id: string;
-  threadId: string;
-  labelIds: string[];
-  snippet: string;
-  sizeEstimate: number;
-  raw: string;
-  historyId: string;
-  internalDate: string;
+import { gmail_v1 } from 'googleapis';
+
+export type GmailMessageParsedResponse = gmail_v1.Schema$Message & {
```

Comment:
I'm not a big fan of having errors mixed with the message. Also, it seems that the new type is having all fields as optionals. Can we infer a type from gmail_v1.Schema$Message with all fields we need required?

Do we need to have errors here or can we add it as an additional parameter of functions when needed?

Reply:
Not sure, it was there before, I assume perhaps because of the way how `fetch-by-batch.service.ts` is implented? 🤔

Reply:
aren't you adding the errors to this type?

Reply:
>Do we need to have errors here or can we add it as an additional parameter of functions when needed?

I tried to have a look at this once again and even though im still not completely able to understand what `fetch-by-batch.service.ts` it apparently adds the `error` property to the parsed object when parsing the response here https://github.com/twentyhq/twenty/blob/main/packages/twenty-server/src/modules/messaging/services/fetch-by-batch/fetch-by-batch.service.ts#L107

We can of course make the type nicer by making it a literal like `gmail_v1.Schema$Message | GmailError` but just imagine looping through such an array. Currently it checks whether the `error` property exists and returns early if it does https://github.com/twentyhq/twenty/blob/main/packages/twenty-server/src/modules/messaging/services/fetch-messages-by-batches/fetch-messages-by-batches.service.ts#L70-L74

>Can we infer a type from gmail_v1.Schema$Message with all fields we need required?

I mean, this is the best type we can get, or is it not? It comes straight from Google that clearly says some properties might are optional. When testing, those we need were always defined, but isnt it better to be safe than sorry? Im doing a runtime assertion using:
```
assert(id);
assert(threadId);
assert(historyId);
assert(internalDate);
```
This would throw early (message wouldnt be parsed) if some of them are actually missing. Safer approach actually rather than letting it throw further down the line with "cant read property X of undefined" or saving invalid data into the DB.

Let me know what you think about these two topics :)

Reply:
I prefer the assert pattern that you are using

I've just remove the errors from your type and it looks good